packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -623,6 +626,12 @@ exports.createAssetNodes = ({
       internal: {
         type: `${makeTypeName(`Asset`)}`,
       },
+      sys: {}
+    }
+    
+        // Revision applies to entries, assets, and content types
+    if (assetItem.sys.revision) {
```

Comment:
How can this be true? 🤔

Reply:
Ohh sorry,  `assetItem` !== `assetNode`


But we should move revision out of sys as we extract all properties from sys into the root of the node above.

Reply:
I agree, however I copied this code from above to be consistent with what is already defined for Entry nodes.  If we went back and changed it for Entry nodes, that would be a breaking change.

In my personal opinion, I think they should be in both places.  If someone defines a content type that has a text field named `revision`, then that would (appropriately) override the `revision` attribute that came from sys.  I think that is good behavior, but I would also like to be able to access it via `{ sys { revision } }` in that case.

However, the above is just a hypothetical.  We can move it to the root and just see if anyone has that problem and wants to complain :)

Reply:
Alright thanks! We will keep it this way in this PR, later on in a breaking release we gonna change the structures of entries and assets together.